Add support for the x64 platform in wixproj templates

lb90 opened the pull request:
Just a proposal...

Currently the wixproj templates only support the x86 platform. It's easy to add support for the x64 platform with some manual editing, but having that ready out of the box would be great.

I'm not sure if something can be done better

lb90 committed:
x64 platform in Wix MSBuild templates

dnfclas commented:
[![CLA assistant check](https://cla.dotnetfoundation.org/pull/badge/signed)](https://cla.dotnetfoundation.org/wixtoolset/VisualStudioExtension?pullRequest=17) <br/>All CLA requirements met.

rseanhall commented on src/Templates/v3/Projects/CustomActionCPP/CustomAction.vcxproj:
```diff
@@ -75,6 +108,27 @@
+		</ClCompile>
+		<Link>
+			<AdditionalDependencies>msi.lib;dutil.lib;wcautil.lib;Version.lib;%(AdditionalDependencies)</AdditionalDependencies>
+			<AdditionalLibraryDirectories>$(WIX)sdk\$(WixPlatformToolset)\lib\x86;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
```
This needs to be x64.

lb90 replied:
Good catch! Fixed.

rseanhall commented on src/Templates/v3/Projects/CustomActionCPP/CustomAction.vcxproj:
```diff
@@ -98,6 +152,29 @@
+		</ClCompile>
+		<Link>
+			<AdditionalDependencies>msi.lib;dutil.lib;wcautil.lib;Version.lib;%(AdditionalDependencies)</AdditionalDependencies>
+			<AdditionalLibraryDirectories>$(WIX)sdk\$(WixPlatformToolset)\lib\x86;%(AdditionalLibraryDirectories)</AdditionalLibraryDirectories>
```
This needs to be x64.

lb90 replied:
Fixed

rseanhall reviewed (changes requested):
Thanks. This seems to be the right way to do this.

lb90 committed:
Fixes, more project templates

lb90 commented:
Added more project templates, should be complete by now.

Also added a .wxi include file to the MSI Project template. This is to help with installing in either ProgramFilesFolder or ProgramFiles64Folder, depending on the active Platform configuration. It contains a simple 'PlatformProgramFilesFolder' define. That include file is named 'Platform.wxi', but perhaps we could change the name to 'TargetArch.wxi' or 'TargetPlatform.wxi', any suggestion about naming is welcome.

Thanks!

rseanhall commented on src/Templates/v3/Projects/WixBundleProject/BundleProject.wixproj:
```diff
@@ -10,13 +10,22 @@
+		<OutputPath>bin\$(Platform)\$(Configuration)\</OutputPath>
+		<IntermediateOutputPath>obj\$(Platform)\$(Configuration)\</IntermediateOutputPath>
+	</PropertyGroup>
+	<PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|x64' ">
+		<OutputPath>bin\$(Platform)\$(Configuration)\</OutputPath>
+		<IntermediateOutputPath>obj\$(Platform)\$(Configuration)\</IntermediateOutputPath>
+		<DefineConstants>Debug</DefineConstants>
+	</PropertyGroup>
+	<PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|x64' ">
+		<OutputPath>bin\$(Platform)\$(Configuration)\</OutputPath>
+		<IntermediateOutputPath>obj\$(Platform)\$(Configuration)\</IntermediateOutputPath>
```
Bundles are always x86 in v3, so I don't think we should add an x64 configuration.

rseanhall commented on src/Templates/v3/Projects/WixLibrary/setuplibrary.wixproj:
```diff
@@ -10,13 +10,22 @@
+		<OutputPath>bin\$(Platform)\$(Configuration)\</OutputPath>
+		<IntermediateOutputPath>obj\$(Platform)\$(Configuration)\</IntermediateOutputPath>
+	</PropertyGroup>
+	<PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|x64' ">
+		<OutputPath>bin\$(Platform)\$(Configuration)\</OutputPath>
+		<IntermediateOutputPath>obj\$(Platform)\$(Configuration)\</IntermediateOutputPath>
+		<DefineConstants>Debug</DefineConstants>
+	</PropertyGroup>
+	<PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|x64' ">
+		<OutputPath>bin\$(Platform)\$(Configuration)\</OutputPath>
+		<IntermediateOutputPath>obj\$(Platform)\$(Configuration)\</IntermediateOutputPath>
```
I don't think library projects should have multiple platforms. Best practice is to build one .wixlib with all the architectures, with different `Id`s based on the target architecture.

rseanhall commented on src/Templates/v3/Projects/WixProject/Product.wxs:
```diff
@@ -13,7 +13,7 @@
 	<Fragment>
 		<Directory Id="TARGETDIR" Name="SourceDir">
-			<Directory Id="ProgramFilesFolder">
+			<Directory Id="$(var.PlatformProgramFilesFolder)">
```
I don't think we should do this. You shouldn't build the project as x86 and as x64 and then distribute them both. You would need to vary other things as well, such as the UpgradeCode.

rseanhall commented on src/Templates/v4/Projects/WixBundleProject/BundleProject.wixproj:
```diff
@@ -11,13 +11,22 @@
-		<OutputPath>bin\$(Configuration)\</OutputPath>
-		<IntermediateOutputPath>obj\$(Configuration)\</IntermediateOutputPath>
+		<OutputPath>bin\$(Platform)\$(Configuration)\</OutputPath>
+		<IntermediateOutputPath>obj\$(Platform)\$(Configuration)\</IntermediateOutputPath>
```
I don't want to touch any of the v4 templates. The projects will look completely different.

AraHaan commented:
What about arm64 to the templates as well?

robmen commented:
Interesting proposal. Feel free to send a completed solution in the future. For now, closing the stale PR.
